crates/miden-objects/src/block/proposed_block.rs

```diff
@@ -130,13 +130,9 @@ impl ProposedBlock {
         batches: Vec<ProvenBatch>,
         timestamp: u32,
     ) -> Result<Self, ProposedBlockError> {
-        // Check for empty or duplicate batches.
+        // Check for duplicate and max number of batches.
```

Comment:
Non-PR question, but why does the protocol care about defining maximum counts? Feels like something that should be configurable - and while the node can enforce its own limits, this still sets the upper cap.

Reply:
Yeah, good question. Some kind of ProtocolConfig would be better I guess. I guess right now it makes it easier to verify consistency, for example: The max number of inputs per batch and the max number of batches per block (1024 * 64 = 2^16) guarantees we can use a `BlockNoteTree` of depth 16 to represent all notes created in a block.

But we could verify such invariants in a `ProtocolConfig::new` constructor, and then miden-base could assume it is consistent.